leash.go

```diff
@@ -218,13 +226,21 @@ func modifyEventContents(toBeSent chan event.Event, options GlobalOptions) chan
 
 // dropEventField drops any fields that are to be dropped, drop them before
 // passing the event on down the line to the next consumer
-func dropEventField(field string, toBeSent chan event.Event) chan event.Event {
-	newSent := make(chan event.Event)
+func dropEventField(field string, toBeSent chan event.Event, options GlobalOptions) chan event.Event {
```

Comment:
Hokay. Remind me why this needed to be written as a transformation on a channel? There's no reason that `dropEventField` / `scrubEventField` / `addEventField` / `requestShape` need to be serialized like that. They **do** probably need to operate on the same event, within a single goroutine... but that seems simple enough to do.

I don't think making this more complicated is the answer.